### PR TITLE
Kotlin Archetype: Update versions. Remove unnecessary dependencyManagement

### PR DESCRIPTION
**Pull request by brunoborges**

### CODE REVIEW

File: azure-functions-kotlin-archetype/pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.microsoft.azure</groupId>
     <artifactId>azure-functions-kotlin-archetype</artifactId>
-    <version>1.21-SNAPSHOT</version>
+    <version>1.31-SNAPSHOT</version>
```
Comment (Flanker32) [inline azure-functions-kotlin-archetype/pom.xml]: Archetype version didn't follow semantic version, it should be 1.22-SNAPSHOT for next version.
By the way, should Kotlin archetype keep the same version with java archetype?

Comment (brunoborges) [reply azure-functions-kotlin-archetype/pom.xml]: @Flanker32 I think it should. What's most important is the version of the Azure Functions Java library, and Maven plugin, not the Kotlin version, which can be updated by the user if the user wants. 

We should definitely try to keep the Java and Kotlin archetypes aligned with the Functions library/plugin versions.

Will move to 1.22-SNAPSHOT.

**Comment by brunoborges** [comment]
@jdneo can you please merge this? the artifact is on its way to maven central with version `1.23`

**Comment by jdneo** [comment]
@Flanker32 please help take a look. Thank you.

**Comment by Flanker32** [comment]
@brunoborges  Sorry for the delay, approved.

**Pull request merged**